main.js

```diff
@@ -2900,6 +2900,34 @@ function startInstance(id, wakeUp) {
         }
     }
 
+    // check how much memory is left and log a warning error/if its critical
+    let availableMemMB;
+
+    if (fs.existsSync('/proc/meminfo')) {
+        // on linux we read mem available
+        try {
+            const text = fs.readFileSync('/proc/meminfo', 'utf8');
+            const m = text && text.match(/MemAvailable:\s*(\d+)/);
+            if (m && m[1]) {
+                availableMemMB =  Math.round(parseInt(m[1], 10) * 0.001024); // convert to MB
+            }
+        } catch (err) {
```

Comment:
should we not also use os.freemem as fallback in exception case?

Reply:
or maybe do not use "else" but check if availableMem is still empty ... this would also cover if the "if (m && m[1])" case is false

Reply:
I would not fallback because this will lead to false positives on Linux if we are really not able to determine men available, because free mem < available mem

Reply:
Ok but even then the "if < 100/50" checks need to be guarded by "it is a number" and so we need to ignore all cases where we got no number, or ?!

Reply:
if https://github.com/ioBroker/ioBroker.js-controller/pull/742/files#diff-7a9076d6d94e62c13d641aa71f19ae8eR2912 works it will be a number in every case, if the line does not apply it will be undefined. `undefined < 50 or 100` is false

Reply:
But if you want to we can make it more clear.. code-wise it would work like that

Reply:
aaaahh ok ... I thought undefined would be cased to a 0 ;-) Then ok for me

Reply:
I would not use implied behavior in code without at least a two-line comment explaining it. Rather add undefined explicitly in the checks.

Reply:
Ingo's comment is exactly why :)

Reply:
;-) fair